### PR TITLE
app-portage/repoman: EAPI bump, remove pypy and fix python dependencies

### PR DESCRIPTION
**Pull request by MathyV**
app-portage/repoman: EAPI bump, remove pypy and fix python dependencies

Since repoman depends on lxml, and lxml is not available for pypy, remove it from the compatibility list.

Also added PYTHON_USEDEP to catch this in the future.

Also some cleanup

### CODE REVIEW

File: app-portage/repoman/repoman-2.3.0-r2.ebuild [hunk 1]
```diff
@@ -21,13 +17,11 @@ KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~s
...
 
-DEPEND="dev-lang/python-exec:2"
-
-RDEPEND="
-	dev-lang/python-exec:2
-	!<sys-apps/portage-2.3.0_rc
-	>=dev-python/lxml-3.6.0
-	"
+DEPEND="
+	!<sys-apps/portage-2.3.0_rc[${PYTHON_USEDEP}]
```
Comment (Arfrever) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: `[${PYTHON_USEDEP}]` should not be used in blocker.
`app-portage/repoman` built with `PYTHON_TARGETS="python3_5"` would still collide with `sys-apps/portage-2.2.28` built with `PYTHON_TARGETS="python3_4"` (at least in `/usr/bin/repoman`).

Use `!<sys-apps/portage-2.3.0_rc`.

Comment (MathyV) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: You are correct of course, I was a bit enthusiastic, fixed now.

File: app-portage/repoman/repoman-2.3.0-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	done
+}
+
+TARBALL_PV=${PV}
+SRC_URI="mirror://gentoo/${PN}-${TARBALL_PV}.tar.bz2
+	$(prefix_src_archives ${PN}-${TARBALL_PV}.tar.bz2)
+	https://gitweb.gentoo.org/proj/portage.git/patch/?id=ef33db45a0c1d462411d4ced1857a322c0ab28f6 -> repoman-2.3.0-bug-586864.patch"
+
+python_prepare_all() {
+	epatch "${DISTDIR}/repoman-2.3.0-bug-586864.patch"
```
Comment (SoapGentoo) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: put this in `PATCHES`

File: app-portage/repoman/repoman-2.3.0-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	distutils-r1_python_prepare_all
+
+	if [[ -n ${EPREFIX} ]] ; then
+		einfo "Prefixing shebangs ..."
+		while read -r -d $'\0' ; do
```
Comment (SoapGentoo) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: read into `file`, declare file `local` before

File: app-portage/repoman/repoman-2.3.0-r2.ebuild [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+			local shebang=$(head -n1 "$REPLY")
```
Comment (SoapGentoo) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: `$file` and not `$reply`

File: app-portage/repoman/repoman-2.3.0-r2.ebuild [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+			if [[ ${shebang} == "#!"* && ! ${shebang} == "#!${EPREFIX}/"* ]] ; then
+				sed -i -e "1s:.*:#!${EPREFIX}${shebang:2}:" "$REPLY" || \
```
Comment (SoapGentoo) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: `$file`

File: app-portage/repoman/repoman-2.3.0-r2.ebuild [hunk 6]
```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{4,5} )
+PYTHON_REQ_USE='bzip2(+)'
+
+inherit distutils-r1 eutils multilib
```
Comment (SoapGentoo) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: with `PATCHES`, you can get rid of `eutils` and `multilib`

Comment (MathyV) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: fixed

File: app-portage/repoman/repoman-2.3.0-r2.ebuild (hunk 2 above)
Comment (MathyV) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: fixed

File: app-portage/repoman/repoman-2.3.0-r2.ebuild (hunk 3 above)
Comment (MathyV) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: fixed

File: app-portage/repoman/repoman-2.3.0-r2.ebuild (hunk 5 above)
Comment (MathyV) [inline app-portage/repoman/repoman-2.3.0-r2.ebuild]: fixed

**Comment by SoapGentoo** [comment]
@dol-sen ack?

**Comment by dol-sen** [comment]
looks good
Thank you

**Comment by SoapGentoo** [comment]
@MathyV thanks for the PR :smile:

**Pull request closed without merging**